engine/spec/components/citizens_advice_components/table_spec.rb

```diff
@@ -1,55 +1,79 @@
 # frozen_string_literal: true
 
 RSpec.describe CitizensAdviceComponents::Table, type: :component do
-  subject(:component) do
-    render_inline(
-      described_class.new(
-        header: header,
-        rows: rows,
-        caption: caption.presence
-      )
-    )
+  let(:sample_header) do
+    ["Your location", "Post box collection times"]
   end
 
-  let(:header) do
+  let(:sample_rows) do
     [
-      "Your location",
-      "Post box collection times"
+      ["City or town", "9am to 6.30pm"],
+      ["Areas with lots of businesses - known as commercial", "9am to 7.30pm"],
+      ["Very rural areas - for example, where there aren't many people", "9am to 4pm"],
+      ["Rest of the UK", "9am to 5.30pm"]
     ]
   end
 
-  let(:rows) do
-    [
-      [
-        "City or town",
-        "9am to 6.30pm"
-      ],
-      [
-        "Areas with lots of businesses - known as commercial",
-        "9am to 7.30pm"
-      ],
-      [
-        "Very rural areas - for example, where there aren't many people",
-        "9am to 4pm"
-      ],
-      [
-        "Rest of the UK",
-        "9am to 5.30pm"
-      ]
-    ]
+  context "when missing headers" do
+    subject(:component) do
+      described_class.new(header: [], rows: sample_rows)
+    end
+
+    it "does not render" do
+      render_inline component
+      expect(rendered_component).to have_no_table
```

Comment:
Is `rendered_component` an automagic shorthand thing?

Reply:
Yep, although not especially magic. It's a reader for the output of `render_inline` provided by `ViewComponent::TestHelpers`. https://viewcomponent.org/guide/testing.html#rspec-configuration